fsm/config.go

```diff
@@ -0,0 +1,118 @@
+package fsm
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+// Config contains the states, commands, functions and
+// default messages of the FSM
+type Config struct {
+	States    []string   `yaml:"states"`
+	Commands  []string   `yaml:"commands"`
+	Functions []Function `yaml:"functions"`
+	Defaults  Defaults   `yaml:"defaults"`
+}
+
+// Function lists the transitions available for the FSM
+type Function struct {
+	Transition Transition `yaml:"transition"`
+	Command    string     `yaml:"command"`
+	Slot       Slot       `yaml:"slot"`
+	Extension  string     `yaml:"extension"`
+	Message    []Message  `yaml:"message"`
```

Comment:
Message is now well defined. You cannot mix and match formats anymore, it has to be...

```
message:
  - text: "foo"
  - text: "bar"
  - image: "https://foo/bar/image"
```

I did this because passing around an interface everywhere made it super difficult to work with this. Feel free to raise concerns here.

Reply:
Ok but you can still have an image with text, right? Like this:

```yaml
message:
  - text: "first message, only text"
  - text: "second message, with image"
    image: "https://foo/bar/image"
```

Reply:
Yep that is still supported.